agent/src/main/scala/za/co/absa/atum/agent/AtumContext.scala

```diff
@@ -52,7 +53,14 @@ class AtumContext private[agent] (
    * @return the sub-partition context
    */
   def subPartitionContext(subPartitions: AtumPartitions): AtumContext = {
-    agent.getOrCreateAtumSubContext(atumPartitions ++ subPartitions)(this)
+    val newPartitions = subPartitions.foldLeft(atumPartitions) { case (acc, (k, v)) =>
```

Comment:
Just an idea. Wouldn't intersection be clearer, what's being checked?
```scala
val overlap = atumPartitions.keys.intersect(subPartitions.keys)
if (overlap.nonEmpty) {
  throw PartitioningUpdateException(s"Partition keys '$overlap' already exist. Updates are not allowed.")
}
```

Reply:
okok, refactored